source/Missions/Patch/Agents/AgentInteractionPatch.cs

```diff
@@ -2,26 +2,51 @@
 using Missions.Extensions;
 using SandBox.Conversation.MissionLogics;
 using System;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Conversation;
+using TaleWorlds.Library;
 using TaleWorlds.MountAndBlade;
 
 namespace Coop.Mod.Patch.Agents
 {
     [HarmonyPatch(typeof(MissionConversationLogic), "OnAgentInteraction")]
     public class AgentInteractionPatch
     {
-        public static event Action<Agent, Agent> OnAgentInteraction;
-        static bool Prefix(ref Agent userAgent, ref Agent agent)
+        public static bool Prefix(ref Agent userAgent, ref Agent agent)
         {
             if (!agent.IsNetworkAgent())
             {
+                ProcessSentencePatch.SetInteractedAgents(null, null);
                 return true;
             }
+            
+            ProcessSentencePatch.SetInteractedAgents(userAgent, agent);
+            return true;
 
-            // TODO fix board games
-            //OnAgentInteraction?.Invoke(userAgent, agent);
+        }
+    }
 
-            return false;
+    [HarmonyPatch(typeof(ConversationManager), "ProcessSentence")]
+    public class ProcessSentencePatch
```

Comment:
Should this be in it's own file like the other patches? Or maybe the file renamed to be `AgentInteractionPatches` like the BoardGame ones?